watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -0,0 +1,568 @@
+
+    # nanofiltration
+    dye_sep.nanofiltration.load_parameters_from_database(use_default_removal=True)
+
+    # nf pump
+    dye_sep.P1.load_parameters_from_database(use_default_removal=True)
+    dye_sep.P1.applied_pressure.fix(
+        dye_sep.nanofiltration.applied_pressure.get_values()[0]
+    )
+    dye_sep.P1.lift_height.unfix()
+
+    # desalination
+    desal.P2.efficiency_pump.fix(0.80)
+    operating_pressure = 70e5 * pyunits.Pa
+    desal.P2.control_volume.properties_out[0].pressure.fix(operating_pressure)
+    desal.RO.A_comp.fix(4.2e-12)  # membrane water permeability
+    desal.RO.B_comp.fix(3.5e-8)  # membrane salt permeability
+    desal.RO.channel_height.fix(1e-3)  # channel height in membrane stage [m]
+    desal.RO.spacer_porosity.fix(0.97)  # spacer porosity in membrane stage [-]
+    desal.RO.permeate.pressure[0].fix(pressure)  # atmospheric pressure [Pa]
+    desal.RO.velocity[0, 0].fix(0.25)
+    desal.RO.recovery_vol_phase[0, "Liq"].fix(0.5)
+    m.fs.tb_nf_ro.properties_out[0].temperature.fix(temperature)
+    m.fs.tb_nf_ro.properties_out[0].pressure.fix(pressure)
+
+    # pressure exchanger
+    desal.PXR.efficiency_pressure_exchanger.fix(0.95)
+    # booster pump
+    desal.P3.efficiency_pump.fix(0.80)
+
+    return
+
+
+def initialize_system(m):
+    prtrt = m.fs.pretreatment
+    dye_sep = m.fs.dye_separation
+    desal = m.fs.desalination
+
+    # initialize feed
+    solve(m.fs.feed)
+
+    # pretreatment
+    propagate_state(m.fs.s_feed)
+    s = SequentialDecomposition()
+    s.options.tear_set = []
+    s.options.iterLim = 1
+    s.run(prtrt, lambda u: u.initialize())
+
+    # initialized nf
+    propagate_state(prtrt.s01)
+    seq = SequentialDecomposition()
+    seq.options.tear_set = []
+    seq.options.iterLim = 1
+    seq.run(dye_sep, lambda u: u.initialize())
+
+    # initialize ro
+    propagate_state(m.fs.s_nf)
+    propagate_state(m.fs.s_ro)
+    propagate_state(desal.s01)
+    propagate_state(m.fs.s_disposal)
+    propagate_state(m.fs.s_permeate)
+
+    m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "H2O"] = value(
+        m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["H2O"]
+    )
+    m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "TDS"] = value(
+        m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["tds"]
+    ) + value(m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["dye"])
+
+    desal.RO.feed_side.properties_in[0].flow_mass_phase_comp["Liq", "H2O"] = value(
+        m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "H2O"]
+    )
+    desal.RO.feed_side.properties_in[0].temperature = value(
+        m.fs.tb_nf_ro.properties_out[0].temperature
+    )
+    desal.RO.feed_side.properties_in[0].pressure = value(
+        desal.P2.control_volume.properties_out[0].pressure
+    )
+    solve(desal)
+    desal.RO.initialize()
+    return
+
+
+def solve(blk, solver=None, tee=False, check_termination=True):
```

Comment:
Technically, since we have the RO unit here, we could have a `setup_optimization` function which unfixes certain variables on the RO unit and then solve and find the cost-optimal decision variables (e.g., membrane area)

Reply:
(just food for thought rather than a suggested change)